src/test/mockFactory.ts

```diff
@@ -0,0 +1,80 @@
+import Serverless from 'serverless';
+import Service from 'serverless/classes/Service';
+import Utils = require('serverless/classes/Utils');
+import PluginManager = require('serverless/classes/PluginManager');
+
+export class MockFactory {
+  public static createTestServerless(service?: Service, utils?: Utils, pluginManager?: PluginManager): Serverless {
+    const sls = new Serverless();
```

Comment:
We should expose the serverless `config` param in this mock factory that gets passed into `Serverless` ctor.